### ISSUE TITLE
Command line processing causing exception

### ISSUE THREAD

**Issue by jesperavot** [piwik/plugin-QueuedTracking#12]
Hi,

I'm running into an issue with the current version of the plugin. I configured Piwik 2.14.0-rc1 and QueuedTracking version 0.1.6, setup Redis and the plugin configuration and noticed that everything is being collected perfectly. However, once I switch Process during tracking request off and try to process the queue using the console, I get the following exception:

**[Piwik\Container\ContainerDoesNotExistException] The root container has not been created yet.**

Looking at the code, it seems it is caused in Factory.php *(StaticContainer::get('Piwik\Plugins\QueuedTracking\Settings');)* because all the other commands work as expected like queuedtracking:monitor

Cheers,
Jesper

**Comment by centminmod** [comment]
same experience as well

**Comment by ldidry** [comment]
Migrated to Piwik 2.14 right now: same error. :frowning:

**Comment by mattab** [comment]
Hi @tsteur! could you take a look at this issue?

**Comment by tsteur** [comment]
Or maybe @diosmosis has an idea why it maybe fails around here https://github.com/piwik/plugin-QueuedTracking/blob/master/Commands/Process.php#L43 ? This is - I think - the only difference to the monitor command

**Comment by diosmosis** [comment]
@tsteur `recreateEagerCacheInstanceWhichChangesOnceTrackerModeIsEnabled()` will clear the container but doesn't initiate a new environment. Which you shouldn't do either. Instead of loading the tracker plugins / environment explicitly, should create and init a `new Environment('tracker');` instance.

I might be able to fix it today, not sure if there are tests for this though.

**Comment by ldidry** [comment]
@diosmosis any news on your fix ?

**Comment by samgabriel** [comment]
For those of you who are still stranded after this bug. Add these two lines to the recreateEagerCacheInstanceWhichChangesOnceTrackerModeIsEnabled 
after 
   ```
     StaticContainer::clearContainer();
        Log::unsetInstance();
```
add
```
        $environment = new \Piwik\Application\Environment('tracker');
		$environment->init();
```
The full function should look like this:
```
private function recreateEagerCacheInstanceWhichChangesOnceTrackerModeIsEnabled()
    {
        StaticContainer::clearContainer();
        Log::unsetInstance();
        $environment = new \Piwik\Application\Environment('tracker');
		$environment->init();

        $key = 'Piwik\Cache\Eager';
        $container = StaticContainer::getContainer();
        $container->set($key, $container->make($key));
    }
```

**Comment by diosmosis** [comment]
I created a fix in #17, but I'm not setup to test it. Hopefully @tsteur will verify soon whether it works
